planning/motion_velocity_smoother/src/trajectory_utils.cpp

```diff
@@ -212,22 +212,17 @@ std::vector<double> calcTrajectoryCurvatureFrom3Points(
 
   // calculate curvature by circle fitting from three points
   std::vector<double> k_arr;
-  for (size_t i = idx_dist; i < trajectory.size() - idx_dist; ++i) {
-    try {
-      const auto p0 = getPoint(trajectory.at(i - idx_dist));
-      const auto p1 = getPoint(trajectory.at(i));
-      const auto p2 = getPoint(trajectory.at(i + idx_dist));
-      k_arr.push_back(calcCurvature(p0, p1, p2));
-    } catch (...) {
-      k_arr.push_back(0.0);  // points are too close. No curvature.
-    }
+  //  for first point curvature = 0;
+  k_arr.push_back(0.0);
+  for (size_t i = 1; i + 1 < trajectory.size(); i++) {
+    const auto p0 = getPoint(trajectory.at(i - std::min(idx_dist, i)));
+    const auto p1 = getPoint(trajectory.at(i));
+    const auto p2 = getPoint(trajectory.at(i + std::min(idx_dist, trajectory.size() - 1 - i)));
+    k_arr.push_back(calcCurvature(p0, p1, p2));
```

Comment:
The calcCurvature() in autoware_utils lib throws an exception when these points are too close. Would you add special handling for closed points? Maybe use previous curvature or something.
(Or just keep using try-catch. It is not a nice way though).

Reply:
I added the error handler, if there is a previous curvature, it push back. Else it push back 0.0. Is it proper way to do?

Reply:
LGTM. Thank you!